packages/pwa-kit-react-sdk/src/ssr/browser/main.jsx

```diff
@@ -33,6 +33,36 @@ export const registerServiceWorker = (url) => {
     })
 }
 
+const OuterApp = ({routes, error, WrappedApp, locals}) => {
+    const AppConfig = getAppConfig()
+    const isInitialPageRef = useRef(true)
+
+    return (
+        <ServerContext.Provider value={{}}>
+            <Router>
+                <CorrelationIdProvider correlationId={() => {
+                    // If we are hydrating an error page use the server correlation id.
+                    if (isInitialPageRef.current && window.__ERROR__) {
+                        isInitialPageRef.current = false
+                        return window.__INITIAL_CORRELATION_ID__
+                    }
```

Comment:
For my own eduction here, this is another one of our tricks to have various React lifecycle things be aware of first (SSR) vs. second (CSR) render passes?

Reply:
I wouldn't say this is something in the bag of tricks. The reason I used `useRef` here was because the alternative is using a global variable. The solution here is pretty much specific to the problem at hand, which is that we don't want to generated a new CSR correlation id when the thing we are rendering is an error page that happened on the server.